src/Arcus.Security.Providers.AzureKeyVault/Authentication/ManagedServiceIdentityAuthenticator.cs

```diff
@@ -8,17 +8,17 @@ namespace Arcus.Security.Providers.AzureKeyVault.Authentication
     /// <summary>
     ///     Azure Key Vault authentication by using Azure Managed Service Identity
     /// </summary>
-    public class ManagedServiceIdentityAuthenticator : IKeyVaultAuthenticator
+    public class ManagedServiceIdentityAuthenticator : IKeyVaultAuthentication
```

Comment:
You'll need to implement both actually, otherwise, this is a breaking change

Reply:
Ah, yes; hmm; then I'll need to rename `Authenticate`.

Reply:
Renamed it to `AuthenticateClient`

Reply:
Noticed this, don't really see the need why we need to rename it to be honest.

Idea is that it authenticates, doesn't need to imply that it gives you a client imo.

I think we should keep `Authenticate`

Reply:
Yes, changed.